<launch>

  <arg name="frame_id" default="/kinect_optical_frame"/>

  <machine name="localhost" address="localhost" env-loader="/opt/ros/hydro/env.sh"/>
  <node pkg="nodelet" type="nodelet" name="kinect_nodelet_manager" args="manager"
      output="screen" machine="localhost" />

  <include file="$(find state_manager)/launch/stateManager.launch" />

  <include file="$(find pandora_vision_hole)/launch/pandora_vision_hole.launch"
      ns="/pandora_vision/pandora_vision_hole" >
    <arg name="thermal" value="false" />
    <arg name="rgbd" value="true" />
    <arg name="rgbdt" value="false" />
    <arg name="hole_standalone_nodelet" value="true" />
  </include>

  <include file="$(find pandora_robots_description)/gio/launch/gio_description.launch" />

  <param name="dataset_path" value="/test/benchmark/data/bags_pcl"/>
  <param name="dataset_camera" value="kinect"/>

  <param name="/kinect_optical_frame/hfov" value="58.0" />
  <param name="/kinect_optical_frame/vfov" value="45.0" />


  <param name="benchmarkFlag" value="False" />
  <param name="kinect/topic_name" value="/kinect/depth_registered/points" />
  <param name="publisherMessagePackage" value="sensor_msgs" />
  <param name="publisherMessageType" value="PointCloud2"/>

  <param name="syncSubscriberTopic" value="/pandora_vision/pandora_vision_hole/hole_fusion/processor_log" />
  <param name="subscriberMessagePackage" value="sensor_processor" />
  <param name="subscriberMessageType" value="ProcessorLogInfo"/>

  <param name="subscriberAlertTopic" value="/vision/holes_direction_alert" />
  <param name="subscriberAlertMessagePackage" value="pandora_vision_msgs" />
  <param name="subscriberAlertMessageType" value="HoleDirectionAlertVector"/>

  <param name="algorithm" value="Hole"/>

  <test test-name="HoleBenchmarkTest" pkg="pandora_vision_hole"
    type="hole_benchmark_test.py" time-limit="6000.0"/>


</launch>
